compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -258,7 +258,15 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 Ok(ty::Visibility::Restricted(DefId::local(CRATE_DEF_INDEX)))
             }
             ast::VisibilityKind::Inherited => {
-                Ok(ty::Visibility::Restricted(parent_scope.module.normal_ancestor_id))
+                if self.parent_scope.module.kind.is_enum() {
+                    // Any inherited visibility resolved directly inside an enum (e.g. variants or fields) inherits from the visibility of the enum.
```

Comment:
(Tidy cannot work on rustfmt-controlled code, unfortunately, but keeping 100-character line limit is still reasonable.)